prover/protocol/column/column.go

```diff
@@ -119,7 +120,11 @@ func EvalExprColumn(run ifaces.Runtime, board symbolic.ExpressionBoard) smartvec
 		case ifaces.Column:
 			inputs[i] = m.GetColAssignment(run)
 		case coin.Info:
-			v := run.GetRandomCoinField(m.Name)
+			if m.Type == coin.FromSeed {
+				v = run.GetRandomCoinFromSeed(m.Name)
```

Comment:
It would be simpler if you modified GetRandomCoinField to be able to do that. That way you would not change this everytime we sample a coin.